### PR TITLE
use symbol table for global symbols

### LINKED ISSUE #1401: [Compiler / IL] Current string interner is thread-unsafe
Just documenting an issue we discussed in person earlier.

Introduced in #1310, the IDs currently used by Calyx are very thread unsafe, which means the Calyx IL cannot currently be used in any threaded contexts, though it does not indicate this.

The root issue is that the singleton method allows multiple mutable accesses to the string interner which can result in erratic behavior and wild memory corruption. I.e., does not enforce the exclusivity of the mutable reference.

The solution requires some form of synchronization.

### PR DESCRIPTION
**Pull request by EclecticGriffin**
A quick fix to close #1401. Swaps in `GlobalSymbol` for `GSym` which pretty seamlessly resolves the thread safety issues. We can probably revisit this later if needed. I could see using the `parking_lot` mutexes as a better alternative to the standard library ones since I don't see lock poisoning as useful in a string table context

### CODE REVIEW

File: calyx-utils/Cargo.toml [hunk 1]
```diff
@@ -12,4 +12,5 @@ serde = { workspace = true, features = ["derive"], optional = true }
 atty.workspace = true
 string-interner.workspace = true
 itertools.workspace = true
-petgraph.workspace = true
\ No newline at end of file
+petgraph.workspace = true
+symbol_table = { version = "0.3", features = ["global", "serde"] }
```
Comment (rachitnigam) [inline calyx-utils/Cargo.toml:16]: The `serde` feature should presumably be gated to only activate when `serialize` feature is requested right?

Comment (EclecticGriffin) [reply calyx-utils/Cargo.toml:16]: Oh probably, I'll fix that once I get around to looking up the proper syntax

**Comment by EclecticGriffin** [comment]
This should be all good to go now, just waiting for the green light on tests. Made some minor modifications to the `jq` fud stage to allow passing flags. Also tweaked a line in `calyx_py` that was giving me an error for some reason.

**Comment by sampsyo** [comment]
Awesome!! Nice work tracking down all the places where we wanted `--sort-keys`.

**Pull request merged**